em-hpo-run.sh

```diff
@@ -0,0 +1,246 @@
+                                ;;
+                        iter=*)
+                                TOTAL_ITR=${OPTARG#*=}
+                                ;;
+                        dbtype=*)
+                                DB_TYPE=${OPTARG#*=}
+                                ;;
+                        dbhost=*)
+                                DB_HOST=${OPTARG#*=}
+                                export DB_HOST=${DB_HOST}
+                                ;;
+                        usertunables=*)
+                                OPTIONS_VAR=${OPTARG#*=}
+                                export OPTIONS_VAR=${OPTIONS_VAR}
+                                ;;
+                        connection=*)
+                                CONNECTIONS=${OPTARG#*=}
+                                ;;
+                        *)
+                esac
+                ;;
+
+        s)
+                BENCHMARK_SERVER="${OPTARG}"
+                ;;
+        e)
+                RESULTS_DIR_PATH="${OPTARG}"
+                export RESULTS_DIR=${RESULTS_DIR_PATH}
+                ;;
+        w)
+                WARMUPS="${OPTARG}"
+                ;;
+        m)
+                MEASURES="${OPTARG}"
+                ;;
+        i)
+                TOTAL_INST="${OPTARG}"
+                ;;
+        r)
+                RE_DEPLOY="true"
+                ;;
+        g)
+                TFB_IMAGE="${OPTARG}"
+                ;;
+        n)
+                NAMESPACE="${OPTARG}"
+                ;;
+        t)
+                THREAD="${OPTARG}"
+                ;;
+        d)
+                DURATION="${OPTARG}"
+                ;;
+        esac
+done
+
+if [[ -z "${BENCHMARK_SERVER}" || -z "${RESULTS_DIR_PATH}" ]]; then
+        echo "Do set the variables - BENCHMARK_SERVER and RESULTS_DIR_PATH "
+        usage
+fi
+
+if [ -z "${EXPERIMENT_TRIALS}" ]; then
+        EXPERIMENT_TRIALS=2
+fi
+
+if [ -z "${SLO_OBJ_FUNC}" ]; then
+        SLO_OBJ_FUNC="( 125 \* float(data.split(\" , \")[1]) ) / ( 150 \* float(data.split(\" , \")[2]) ) / ( (25 \* float(data.split(\" , \")[3]) )/100 )"
+fi
+
+if [ -z "${SLO_DATA_ROW}" ]; then
+        SLO_DATA_ROW=4
+fi
+
+if [ -z "${SLO_DIRECTION}" ]; then
+        SLO_DIRECTION="maximize"
+fi
+
+if [ -z "${BENCHMARK_NAME}" ]; then
+        BENCHMARK_NAME="techempower"
+fi
+
+if [ -z "${WARMUPS}" ]; then
+        WARMUPS=5
+fi
+if [ -z "${MEASURES}" ]; then
+        MEASURES=3
+fi
+
+if [ -z "${TOTAL_INST}" ]; then
+        TOTAL_INST=1
+fi
+
+if [ -z "${TOTAL_ITR}" ]; then
+        TOTAL_ITR=1
+fi
+
+if [ -z "${RE_DEPLOY}" ]; then
+        RE_DEPLOY=false
+fi
+
+if [ -z "${TFB_IMAGE}" ]; then
+        TFB_IMAGE="${TFB_DEFAULT_IMAGE}"
+fi
+
+if [ -z "${NAMESPACE}" ]; then
+        NAMESPACE="autotune-tfb"
+fi
+
+if [ -z "${THREAD}" ]; then
+        THREAD="40"
+fi
+
+if [ -z "${DURATION}" ]; then
+        DURATION="60"
+fi
+
+if [ -z "${CONNECTIONS}" ]; then
+        CONNECTIONS="512"
+fi
```

Comment:
Wondering if these would be better if we just pre initialize them before reading the user parameters and based on the user params only the right ones get set.

```
THREAD="40"
...
case
   ...
        t)
                THREAD="${OPTARG}"
                ;;
...
```

Reply:
Done.